include/deal.II/lac/vector.templates.h

```diff
@@ -304,7 +304,7 @@ void Vector<Number>::reinit (const size_type n,
     }
 
   if (omit_zeroing_entries == false)
-    *this = static_cast<Number>(0);
+    *this = Number(0);
```

Comment:
Can't you just say `Number()`? This creates the identity element with respect to addition of the respective number format and should work for double, float, complex, etc. I find that clearer than casting or constructing the 'zero'...